lib/ak_page_creator.rb

```diff
@@ -8,36 +8,45 @@ class << self
     def create_page(name, title, language, url, type, page_id)
       new.create_page(name, title, language, url, type, page_id)
     end
+
+    def page_types
+      SUPPORTED_PAGE_TYPES
+    end
   end
 
   def create_page(name, title, language, url, type, page_id)
     translated_language = AkLanguageUriFinder.get_ak_language_uri(language)
+
     case type
       when SUPPORTED_PAGE_TYPES[:donation]
-        self.process_create_request(
-            self.create_donation_page(name, title, translated_language, url),
-            page_id
+        process_create_request(
+          create_donation_page(name, title, translated_language, url),
+          page_id,
+          type
         )
       when SUPPORTED_PAGE_TYPES[:petition]
-        self.process_create_request(
-            self.create_petition_page(name, title, translated_language, url),
-            page_id
+        process_create_request(
+          create_petition_page(name, title, translated_language, url),
+          page_id,
+          type
         )
-      else
-        # do nothing, at least for now.
     end
   end
 
-  def process_create_request(request, page_id)
+  def process_create_request(request, page_id, type)
     page = Page.find(page_id)
 
-    if request.response.class == Net::HTTPCreated
-      page.status = 'success'
-      page.save
-    elsif request.response.class == Net::HTTPBadRequest
-      page.status = 'failed'
-      page.messages = request.parsed_response.to_json
-      page.save
+    case request.response
```

Comment:
ughh that's so confusing! after reading about `===` I don't want to use statements any more. It's so uninuitive that case doesn't do equality matching!